controllers/src/main/java/com/redhat/ipaas/controllers/integration/online/ActivateHandler.java

```diff
@@ -83,29 +83,54 @@ public StatusUpdate execute(Integration integration) {
 
         // TODO: Verify Token and refresh if expired ....
 
-        Optional<Integration.Status> currentStatus = integration.getCurrentStatus();
-        if (!currentStatus.isPresent() || currentStatus.get() != Integration.Status.Pending) {
-            String gitHubUser = getGitHubUser();
-            log.info("{} : Looked up GitHub user {}", getLabel(integration), gitHubUser);
-
-            Map<String, byte[]> projectFiles = createProjectFiles(gitHubUser, integration);
-            log.info("{} : Created project files", getLabel(integration));
-
-            String gitCloneUrl = ensureGitHubSetup(integration, getWebHookUrl(deployment, secret), projectFiles);
-            log.info("{} : Updated GitHub repo {}", getLabel(integration), gitCloneUrl);
-
-            ensureOpenShiftResources(integration.getName(), gitCloneUrl, secret, extractApplicationPropertiesFrom(integration));
-            log.info("{} : Created OpenShift resources", getLabel(integration));
-        } else {
-            log.info("{} : Skipping project generation because integration is in status {}",
-                     getLabel(integration), currentStatus.map(Enum::name).orElse("[none]"));
+        int currentStep = integration.getCurrentStatusStep().orElse(0).intValue();
+        Integration.Status currentStatus = integration.getCurrentStatus().orElse(Integration.Status.Draft);
+        if( currentStatus != Integration.Status.Pending ) {
+            currentStep = 0;
         }
 
-        Integration.Status status = openShiftService.isScaled(deployment) ?
-            Integration.Status.Activated : Integration.Status.Pending;
-        log.info("{} : Setting status to {}", getLabel(integration), status);
+        try {
+            String gitCloneUrl = null;
+            switch (currentStep) {
+                default:
+                case 0:
+                    String gitHubUser = getGitHubUser();
+                    log.info("{} : Looked up GitHub user {}", getLabel(integration), gitHubUser);
+
+                    Map<String, byte[]> projectFiles = createProjectFiles(gitHubUser, integration);
+                    log.info("{} : Created project files", getLabel(integration));
+
+                    gitCloneUrl = ensureGitHubSetup(integration, getWebHookUrl(deployment, secret), projectFiles);
+                    log.info("{} : Updated GitHub repo {}", getLabel(integration), gitCloneUrl);
+                    currentStep = 1;
+
```

Comment:
it's with intension.  I want it continue onto the next steps. But skip some on a retry.